app/services/api/v1/index_message_service.rb

```diff
@@ -0,0 +1,20 @@
+module Api
+  module V1
+    class IndexMessageService
+      def index(conversation_id, current_user, page)
+        conversation = Conversation.find(conversation_id)
+        raise InvalidConversationError unless conversation.user1_id == current_user.id || conversation.user2_id == current_user.id
+
+        messages = conversation.messages.page(page)
+        read_messages(messages)
+        messages
+      end
+
+      private
+
+      def read_messages(messages)
+        messages.each { |message| message.update(read: true) }
```

Comment:
I think this should update the messages from the opposite user so you don't mark as read the message sent by yourself